rp2040-hal/src/adc.rs

```diff
@@ -92,11 +129,30 @@ impl Adc {
     pub fn disable_temp_sensor(&mut self, _: TempSense) {
         self.device.cs.modify(|_, w| w.ts_en().clear_bit());
     }
+
+    fn read(&mut self, chan: u8) -> u16 {
+        while !self.device.cs.read().ready().bit_is_set() {
+            cortex_m::asm::nop();
+        }
+
+        self.device
+            .cs
+            .modify(|_, w| unsafe { w.ainsel().bits(chan).start_once().set_bit() });
+
+        while !self.device.cs.read().ready().bit_is_set() {
+            cortex_m::asm::nop();
+        }
+
+        self.device.result.read().result().bits()
+    }
 }
 
 macro_rules! channel {
```

Comment:
I think "impl Trait for Type" should be moved to be under `struct Type`. Or possibly under the `trait Trait` block, depending on which makes more sense to the casual observer. But usually the former.